### ISSUE TITLE
As a profile author I want to be able to set dependency ordering between kustomization and helm artifacts

### ISSUE THREAD

**Issue by aclevername** [weaveworks/profiles#215]
**What feature/behavior/change do you want?**
#214 introduced the ability to have artifacts of the same kind depend on each other to determine ordering. This did not allow me to have a kustomization that relies on a helm chart being deployed. I want to be able to do this.

**Why do you want this feature?**
Example: I have a Chart for deploying my CRDs and a kustomization for deploying my instances of that CRD. I need to ensure that the CRD is deployed first.

## Proposal
We do not need to change the current specification, as we already have the required fields in the profile definition spec for specifying `dependsOn` as part of #214 . In order to make it work we have to start wrapping all `HelmRelease` resources in a `Kustomization` resource. The spike https://github.com/weaveworks/profiles/issues/196 outlines how to do this and contains an [example](https://github.com/aclevername/flux2-examples).

 The TLDR is that if you have a kustomiation resource that is responsible for deploying the `yaml` file for the helmrelease resource you can use kustomization resources to enforce dependency ordering. Essentially treating all helm chart artifacts as kustomization artifacts.

**Comment by nikimanoledaki** [comment]
This is sized as M but could be L - whoever picks this up can pair with @aclevername to get more context.

**Comment by aclevername** [comment]
Discussion in flux on adding native support for this https://github.com/fluxcd/flux2/discussions/1599
